openshift/catalog-sync/pkg/check/catalog.go

```diff
@@ -0,0 +1,143 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+func AllCatalogChecks() []CatalogCheck {
+	return []CatalogCheck{
+		BundleHasValidMetadataStructure(),
+		ChannelHeadsHaveCSVMetadata(),
+	}
+}
+func BundleHasValidMetadataStructure() CatalogCheck {
+	return CatalogCheck{
+		Name: "CheckBundleHasValidMetadataStructure",
+		Fn: func(ctx context.Context, fbcFS fs.FS) error {
+			const configsDir = "configs"
+
+			entries, err := fs.ReadDir(fbcFS, configsDir)
+			if err != nil {
+				return fmt.Errorf("read configs dir: %w", err)
+			}
+
+			var errs []error
+
+			for _, entry := range entries {
```

Comment:
This code makes the assumption that catalogs will have subdirectories, each of which contain FBC. We (OLM) have generally not been prescriptive of this hierarchy AND when we've see other pipeline code that assumes this hierarchy, we've pointed it out as potentially problematic.

Since this check is about metadata, and not the directory structure, I think we should `declcfg.Load` the entire `configsDir`, and perform the check from there.